tests/lib/timespec_util/testcase.yaml

```diff
@@ -1,36 +1,91 @@
 # FIXME: this should be under tests/unit/timeutil but will not work due to #90029
 common:
-  filter: not CONFIG_NATIVE_LIBC
   tags:
     - timeutils
   # 1 tier0 platform per supported architecture
   platform_key:
     - arch
     - simulation
   integration_platforms:
+    - native_sim
```

Comment:
This should be added in this PR. Please feel free to add it in a separate follow-up PR after 

https://github.com/zephyrproject-rtos/zephyr/pull/93535 is merged